src/libraries/System.Linq.AsyncEnumerable/src/System/Linq/Except.cs

```diff
@@ -26,27 +26,44 @@ public static IAsyncEnumerable<TSource> Except<TSource>(
             ThrowHelper.ThrowIfNull(first);
             ThrowHelper.ThrowIfNull(second);
 
-            return Impl(first, second, comparer, default);
+            return
+                first.IsEmpty() ? Empty<TSource>() :
+                Impl(first, second, comparer, default);
```

Comment:
no special casing if i second is empty?  is that because tehre might be a different comparer passed along?

Reply:
If the second is empty, we still need to yield elements from the first.

Reply:
yup.  thanks